swarm/storage/ldbstore_test.go

```diff
@@ -584,6 +584,141 @@ func TestLDBStoreCollectGarbageAccessUnlikeIndex(t *testing.T) {
 	log.Info("ldbstore", "total", n, "missing", missing, "entrycnt", ldb.entryCnt, "accesscnt", ldb.accessCnt)
 }
 
+func TestCleanIndex(t *testing.T) {
+	capacity := 5000
+	n := 3
+
+	ldb, cleanup := newLDBStore(t)
+	ldb.setCapacity(uint64(capacity))
+	defer cleanup()
+
+	chunks, err := mputRandomChunks(ldb, n, 4096)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// remove the data of the first chunk
+	po := ldb.po(chunks[0].Address()[:])
+	dataKey := make([]byte, 10)
+	dataKey[0] = keyData
+	dataKey[1] = byte(po)
+	// dataKey[2:10] = first chunk has storageIdx 0 on [2:10]
+	_, err = ldb.db.Get(dataKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ldb.db.Delete(dataKey)
```

Comment:
this returns an error

Reply:
Actually, the underlying method does not.

Reply:
i don't understand...
`func (db *LDBDatabase) Delete(key []byte) error {...`

Reply:
Have a look at `godoc github.com/syndtr/goleveldb/leveldb | less` -> Delete